kthfsdv-plotting/main.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+import matplotlib
+from matplotlib import pyplot as plt
+from matplotlib.widgets import Button, TextBox
+import pandas as pd 
+
+class Exc2():
+	def __init__(self, tmin, tmax, updateTime):
+		self.tmin = tmin
+		self.tmax = tmax
+		self.updateTime = updateTime
+		self.H = []
+		self.T = []
+		self.pause = False
+		self.saveFolder ="data"
+
+	def h_t(self, t):
+		return 3*np.pi * np.exp(-self.lambdafunc(t))
+
+	def lambdafunc(self, t):
+		return 5*np.sin(2*np.pi*t)
+	
+	# Play/Pause button
+	def onClick(self, event):
+		self.pause = False if self.pause == True else True
+	
+	# Reset the plot
+	def onClickReset(self, event):
+		self.H = []
+		self.T = []
+		self.t= self.tmin
+
+	# Edit tmin variable
+	def submitTmin(self, event):
+		self.tmin = int(event)
+	
+	# Edit tmax variable
+	def submitTmax(self, event):
+		self.tmax = int(event)
+
+	# Edit save folder name
+	def submitSave(self, event):
+		self.saveFolder = event
+	
+	# Save to folder
+	def save(self, event):
+		savestring = self.saveFolder + '.csv'
+		arr = np.array([self.H, self.T])
+		np.savetxt(savestring, arr, delimiter=",")
+
+	def plot(self):
+		# Initialize figure
+		plt.ion()
+		fig = plt.figure(figsize=(13,6))
+		ax = fig.add_subplot(111)
+		plt.subplots_adjust(bottom=0.2)
+		ax.plot(self.T, self.H) 
+		plt.ylabel('H')
+		plt.xlabel('t')
+		plt.title('Timeseries')
+
+		# Add buttons and Text Fields
+		axplaypause = plt.axes([0.7, 0.05, 0.1, 0.075])
+		bplaypause = Button(axplaypause, 'Play/Pause')
+		bplaypause.on_clicked(self.onClick)
+		
+		axreset = plt.axes([0.81, 0.05, 0.1, 0.075])
+		breset = Button(axreset, 'Reset')
+		breset.on_clicked(self.onClickReset)
+
+		tminbox = plt.axes([0.3, 0.05, 0.05, 0.075])
+		tmin_box = TextBox(tminbox, 't min', initial=str(self.tmin))
+		tmin_box.on_submit(self.submitTmin)
+
+		tmaxbox = plt.axes([0.4, 0.05, 0.05, 0.075])
+		tmax_box = TextBox(tmaxbox, 't max', initial=str(self.tmax))
+		tmax_box.on_submit(self.submitTmax)
+
+		tsavebox = plt.axes([0.5, 0.05, 0.1, 0.075])
+		tsave_box = TextBox(tsavebox, 'Folder', initial=self.saveFolder)
+		tsave_box.on_submit(self.submitSave)
+
+		axsave = plt.axes([0.6, 0.05, 0.1, 0.075])
+		bsave = Button(axsave, 'Save to Folder')
+		bsave.on_clicked(self.save)
+
+		# Loop through and plot data
+		self.t = self.tmin
+		while self.t < self.tmax:
+			if not self.pause:
+				h = self.h_t(self.t)
+				self.H.append(h)
+				self.T.append(self.t)
+				ax.cla()
+				ax.plot(self.T, self.H)
+				plt.pause(self.updateTime)
+				self.t +=1
```

Comment:
This update is not consistent with the actual `updateTime`. Shouldn't it be the following?
```suggestion
				self.t += self.updateTime
```